drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTablePresenter.java

```diff
@@ -1005,6 +1006,18 @@ private void doDeleteColumn(final BaseColumn column,
         callback.execute();
     }
 
+
+    @Override
+    public void onSort(final GridColumn gridColumn) {
+        try {
+           final List<Integer> sort = uiModel.sort(gridColumn);
+            refreshView();
+            analyzerController.sort(sort);
```

Comment:
Actual sorting is done in uiModel with the help of `TableSortComparator`. That then returns the the new row order with each row number in a list. I chose to do the sort in UIModel since we sort by visible values.

Reply:
thank you